aiida/schedulers/plugins/lsf.py

```diff
@@ -119,13 +119,13 @@ def __init__(self, **kwargs):
 
         try:
             self.parallel_env = str(kwargs.pop('parallel_env', ''))
-        except (TypeError, ValueError):
-            raise TypeError("When specified, 'parallel_env' must be a string")
+        except (TypeError, ValueError) as exc:
+            raise TypeError("When specified, 'parallel_env' must be a string") from exc
```

Comment:
Are we catching anything here? I can't find anything that can't be casted into `str` (see [this](https://stackoverflow.com/questions/38380335/potential-exceptions-using-builtin-str-type-in-python) for example) and the `pop` has a default so...?

Reply:
I think you might be right. The default in `get` should guarantee a value. If your reference is correct that `str()` cannot ever raise, then we would not need to catch any exception. Still I am not a 100% sure and if we missed something and introduce a bug that be kind of annoying

Reply:
So I added a hacky test for this, but thinking a bit more about it...

Should we really be doing implicit validation through this try-catch? Or should do a more explicit input validation for `parallel_env` that raises if anything other than a string is provided? Does it make sense to pass anything else that would get casted, what is this variable for?

Reply:
given the error message, it would make sense to just have something like:
```
if not isinstance(kwargs.pop('parallel_env', ''), str):
    raise TypeError(...)
```
As it stands, the try-catch indeed makes very little sense. I think it would be fine to make that change and then you don't need such a convoluted test setup

Reply:
Ok, change done and test streamlined.

But I'll have you know that test was pure coding art =P